include/seqan3/io/sequence/sequence_file_format_sam.hpp

```diff
@@ -0,0 +1,352 @@
+                                          decltype(std::istreambuf_iterator<char>{})>
+                            {std::istreambuf_iterator<char>{stream},
+                             std::istreambuf_iterator<char>{}};
+        auto stream_it = ranges::begin(stream_view);
+        if(*stream_it != '@') { //Ignore comment lines
+          // ID
+          read_id(stream_view, options, id);
+          // Sequence
+          if constexpr (!detail::decays_to_ignore_v<seq_type>) // sequence
+              read_seq(stream_view,  options, sequence);
+          else
+              read_seq(stream_view,  options, seq_qual);     // seq_qual (possibly std::ignore, too)
+          // Qualities
+         if constexpr (!detail::decays_to_ignore_v<qual_type>) // qualities
+              read_qual(stream_view,  qualities);
+          else
+              read_qual(stream_view, seq_qual);        // seq_qual (possibly std::ignore, too)
+              //TODO read_qual who returns seq_qual with correct output format
+          // make sure "buffer at end" implies "stream at end"
+          if ((std::istreambuf_iterator<char>{stream} == std::istreambuf_iterator<char>{}) &&
+              (!stream.eof()))
+          {
+              stream.get(); // triggers error in stream and sets eof
+          }
+        }
+        else{
+          while(*stream_it != '\n')
+          {++stream_it;}
+        }
+        // Jump over the new line character in the end
+        ++stream_it;
+    }
+    //!\copydoc sequence_file_out_format_concept::write
+    template <typename stream_type,     // constraints checked by file
+              typename seq_type,        // other constraints checked inside function
+              typename id_type,
+              typename qual_type,
+              typename seq_qual_type>
+    void write(stream_type                     & stream,
+               sequence_file_out_options const & options,
+               seq_type                       && sequence,
+               id_type                        && id,
+              // qual_type                      && SEQAN3_DOXYGEN_ONLY(qualities),
+              qual_type                       && qualities,
+               seq_qual_type                  && seq_qual)
+    {
+        static_assert(detail::decays_to_ignore_v<seq_type> || detail::decays_to_ignore_v<seq_qual_type>,
+                      "Either the sequence field, or the seq_qual field need to be set to std::ignore.");
+        static_assert(detail::decays_to_ignore_v<qual_type> || detail::decays_to_ignore_v<seq_qual_type>,
+                      "Either the qualities field, or the seq_qual field need to be set to std::ignore.");
+        ranges::ostreambuf_iterator stream_it{stream};
+        // ID
+        if constexpr (detail::decays_to_ignore_v<id_type>)
+        {
+            throw std::logic_error{"The ID field may not be set to ignore when writing sam files."};
```

Comment:
Are you sure? I think they can be set to `*` and the file will still be valid, or not?